### PR TITLE
Feat(eos_designs): Improve custom python class API for interface descriptions

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Improve custom python class API for interface descriptions

## Related Issue(s)

Part of #3306

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Add new API functions for custom python modules for interface descriptions.
- Use a data class to pass data between AVD and the interface description class.

## How to test

- Replaced existing custom class with a new one using the new API - no changes!
- Replaced existing custom jinja for spines with the old custom python class - no changes!
- No other changes in molecule.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by tgodaA — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        link_peer = str(data.peer).upper()
+        if data.link_type == "underlay_p2p":
+            return f"{self._custom_description_prefix}_P2P_LINK_TO_{link_peer}_{data.peer_interface}"
+
+        if data.link_type == "underlay_l2":
+            return f"{self._custom_description_prefix}_{link_peer}_{data.peer_interface}"
+
+        return ""
+
+    def underlay_port_channel_interfaces(self, data: InterfaceDescriptionData) -> str:
```
Comment (chetryan) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: Shouldn't this be `underlay_port_channel_interface` ?

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml [hunk 2]
```diff
@@ -7,15 +7,10 @@ node_type_keys:
     type: spine
     default_evpn_role: server
     interface_descriptions:
-      # Override interface description templates with our custom templates
-      underlay_ethernet_interfaces: 'custom_templates/interface_descriptions/underlay/ethernet-interfaces.j2'
```
Comment (chetryan) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml:11]: should we be deleting these template files from the molecule test?

**Review by chetryan — COMMENTED** [review]
LGTM - just 2 NIT questions

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml (hunk 2 above)
Comment (tgodaA) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml:11]: Claus used this to test the python module. So we have to remove the jinja reference

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py (hunk 1 above)
Comment (tgodaA) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: @ClausHolbechArista

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: addressed in [21a1542](https://github.com/aristanetworks/ansible-avd/pull/3311/commits/21a1542657f4c1687976d3e54a7f29d78a7b3e71)

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml [hunk 3]
```diff
@@ -7,15 +7,10 @@ node_type_keys:
...
-      underlay_port_channel_interfaces: 'custom_templates/interface_descriptions/underlay/port-channel-interfaces.j2'
-      mlag_ethernet_interfaces: 'custom_templates/interface_descriptions/mlag/ethernet-interfaces.j2'
-      mlag_port_channel_interfaces: 'custom_templates/interface_descriptions/mlag/port-channel-interfaces.j2'
-      connected_endpoints_ethernet_interfaces: 'custom_templates/interface_descriptions/connected_endpoints/ethernet-interfaces.j2'
-      connected_endpoints_port_channel_interfaces: 'custom_templates/interface_descriptions/connected_endpoints/port-channel-interfaces.j2'
```
Comment (tgodaA) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml:16]: Now I understood @chetryan comment. I think we should be deleting the jinja template files. @carlbuchmann @ClausHolbechArista

Comment (tgodaA) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CUSTOM_PYTHON_MODULES_TESTS.yml:16]: The references still exist in `group_vars/CUSTOM_TEMPLATES_TESTs.yml`

So LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**